keg_elements/forms/__init__.py

```diff
@@ -348,11 +348,50 @@ def wrapper(form):
 
 
 class Form(BaseForm):
+    """Base form with a bunch of QoL improvements
+
+    .. note:: Relying on the default field ordering can lead to unintuitive forms. It is possible to
+        override this by adding the ``__field_order`` class attribute. Set this class variable to a
+        tuple or list of field names (addressable via Form._fields['name_of_field']) and the form
+        will render in that order. You must include all the fields, except CSRF.  Forgetting a field
+        or adding one which doesn't exist will cause the form to raise a ``ValueError`` and the form
+        will not be rendered.
+
+    Example ::
+
+        class MyForm(Form):
+            __field_order = ('field1', 'field2',)
+
+            field1 = String('field1_label')  # Note that we don't use the label in the ordering
+            field2 = String()
+
+    """
     def __init__(self, *args, **kwargs):
         super(Form, self).__init__(*args, **kwargs)
         self._form_level_errors = []
         self.after_init(args, kwargs)
 
+    def __iter__(self):
+        # :rage: ... I have no idea why you need the class name, but alas, you do!
+        cls_name = self.__class__.__name__
+        order = getattr(self, '_{}__field_order'.format(cls_name), None)
+
+        if order is None:
+            return super().__iter__()
+
+        has_csrf = hasattr(self, 'csrf_token')
+        order = list(order) + (['csrf_token'] if has_csrf else [])
```

Comment:
NIT: I would expect the csrf token field to be at the top of the form but if you have a preference for it to be at the bottom that's fine